### ISSUE TITLE
Return React Component from itemRenderer

### ISSUE THREAD

**Issue by tiberiumaxim** [orgsync/react-list#145]
I have a complex list of items where each item is a React Component.

```
<ReactList
  type="variable"
  itemRenderer={this.renderItem}
  itemSizeEstimator={this.itemSizeEstimator}
  length={50}
/>
```

```
itemSizeEstimator(index, cache) {
  return 100;
}
```

```
renderItem(index, key) {
  return <TestItem/>;
}
```

```
class TestItem extends React.Component {
	render() {
		return (
			<div>
				TestItem text text
			</div>
		);
	}
}
```

Whenever I run this, I get infinite cycle, browser freezes.
Is returning a React Component not supported?

**Comment by degroote22** [comment]
It is. Check for the actual height of your components. This gets tricky with margins.
If they're all 100px height, use uniform type then.
If you can tell the height of the **variable** element, use itemSizeGetter func instead.

**Comment by tiberiumaxim** [comment]
@degroote22 thanks for replying. It seems like the problem was from the margins of the elements. Not from the height, which is actual variable and can't be calculated (easily). My example doesn't denote this, I know.

But anyway, with margins it goes into infinite loop. Can't it just throw an error or something?

**Comment by degroote22** [comment]
No, it can't. It's just not that simple. React-list is not coded to work with margins. To get to the same effect use a larger wrapping div that has padding.

The problem with margins is if that you have two sibling (one after the other) divs with margin the browser will use the larger margin between them instead of summing them.
So, if you say your element has height 96 plus 2 margin it wont render at 100 height all the times. The first one will render at 100h but the next one will be 98h. Did you get it? I'm sorry but english is not my native language. Check the web for more if you're still confused.

**Comment by tiberiumaxim** [comment]
@degroote22 I understand what you're saying. And yes, that's problematic, but still, you could compare the two siblings' margins and calculate accordingly. Maybe I will have a look and give a contribution to the repo, because I think it's a pretty popular one in React world.

Ayway, until then, I used pseudo elements to mimic margins.
Instead of div `{ margin-bottom: 50px; }` I went for `div:after { content: ''; display: block; height: 50px; }`
